### PR TITLE
Backend/users

### PR DESCRIPTION
**Pull request by vukRadovanovic**
I updated uploadUserProfilePicture to work correctly now.  Evan changed several functions to bring them up to date with our current database structure.  Evan and I updated error logging in LogIn and SignUp.  I fixed a typo in Users.js and Evan moved two functions (emailTaken, usernameTaken) from Users.js to ManageAccount.js, because we both felt they were more fitting in there.

### CODE REVIEW

File: backend/model/Users.js [hunk 1]
```diff
@@ -6,103 +6,222 @@ require("firebase/auth");
...
+    var userDocRef = usersCollection.doc(current_user.uid);
+    var userProfile;
+
+    // access document
+    await userDocRef.get().then(function(userDoc) {
+      if (userDoc.exists) {
+        // set userProfile to retrieved data
+        // not sure this is how to retrieve data
+        userProfile = {
+          username:     userDoc.get("username"),
```
Comment (terryzfeng) [inline backend/model/Users.js:66]: Just curious but userDoc.get("username"), is that the same thing as userDoc.data().username? That's what Tyus and I figured out on Saturday.

File: backend/model/Users.js [hunk 2]
```diff
@@ -6,103 +6,222 @@ require("firebase/auth");
...
-    console.log('user input: ', username);
-    const response = users.where('username', '==', username);
-    const query = await response.get();
-    return !query.empty;
+  uploadUserProfilePicture: async function (current_user, file) {
+    // TODO this function may not work correctly.
+    // create a cloud storage refrence
+    var storageRef = firebase
+      .storage()
+      .ref(user.uid + "/profilePicture/" + file.name);
```
Comment (terryzfeng) [inline backend/model/Users.js:125]: Should we make a folder in storage for users? Not sure how organization of storage is going to be yet

File: backend/model/Users.js [hunk 3]
```diff
@@ -6,103 +6,222 @@ require("firebase/auth");
...
+
+    // save to cloud storage
+    var task = storageRef.put(file);
```
Comment (terryzfeng) [inline backend/model/Users.js:128]: Should add the type metadata along with the file

**Review by terryzfeng — COMMENTED** [review]
Left a comment regarding getting field values in a document but other than that looks good. We probably need to talk a bit regarding firebase storage still.

File: backend/model/Users.js (hunk 1 above)
Comment (evanjserrano) [reply backend/model/Users.js:66]: Both should be fine

Comment (evanjserrano) [reply backend/model/Users.js:66]: https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentSnapshot#data

File: backend/model/Users.js (hunk 2 above)
Comment (evanjserrano) [reply backend/model/Users.js:125]: I think we should. We should write somewhere what our cloud storage organization will be like

**Review by evanjserrano — APPROVED** [review]

**Comment by terryzfeng** [comment]
Other than storage stuff, which is still in progress, I think it's good.

**Review by evanjserrano — APPROVED** [review]

**Pull request merged**